Jenkinsfile

```diff
@@ -0,0 +1,88 @@
+#!groovy
+@Library('Utils')
+import com.redhat.*
+
+properties([disableConcurrentBuilds()])
+
+node {
+    def source = ""
+    def dockerfiles = null
+    String scmUrl = scm.browser.url
+    String scmRef = "${CHANGE_BRANCH}"
+
+    /* Checkout source and find all the Dockerfiles.
+     * This will not include Dockerfiles with extensions. Currently the issue
+     * with using a Dockerfile with an extension is the oc new-build command
+     * does not offer an option to provide the dockerfilePath.
```

Comment:
not ideal, but maybe a way to approach Dockerfile.xxxx files is the '-D' option:
```shell
cat starter-arbitrary-uid/Dockerfile.centos7 | oc new-build . --name=starter-arbitrary-uid --context-dir=starter-arbitrary-uid -D -
```